test/foundry/Deposit.t.sol

```diff
@@ -0,0 +1,389 @@
+        uint256 aliceAssets,
+        uint256 minShareRatio,
+        bytes memory fuzzedReceiptInformation
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyBob = bound(fuzzedKeyBob, 1, SECP256K1_ORDER - 1);
+        address bob = vm.addr(fuzzedKeyBob);
+
+        minShareRatio = bound(minShareRatio, 0, 1e18);
+
+        vm.assume(alice != bob);
+        // Assume that aliceAssets is not 0
+        vm.assume(aliceAssets != 0);
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        vault.grantRole(vault.DEPOSITOR(), alice);
+        vault.grantRole(vault.DEPOSITOR(), bob);
+
+        // Set up the event expectation for DepositWithReceipt
+        vm.expectEmit(false, false, false, true);
+        emit DepositWithReceipt(alice, bob, aliceAssets, aliceAssets, 1, fuzzedReceiptInformation);
+
+        vault.deposit(aliceAssets, bob, minShareRatio, fuzzedReceiptInformation);
+
+        // Assert that the total supply and total assets are equal after the deposit
+        assertEqUint(vault.totalSupply(), vault.totalAssets());
+        vm.stopPrank();
+    }
+
+    /// Test PreviewDeposit returns correct shares
+    function testPreviewDepositReturnedShares(
+        uint256 fuzzedKeyAlice,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 aliceAssets
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        vault.grantRole(vault.DEPOSITOR(), alice);
+        uint256 shares = vault.previewDeposit(aliceAssets);
+
+        assertEqUint(shares, aliceAssets);
+
+        vm.stopPrank();
+    }
+
+    //Test PreviewDeposit returns correct assets
+    function testPreviewMintReturnedAssets(
+        uint256 fuzzedKeyAlice,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 shares
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
```

Comment:
this kind of poisons the test here because you're deploying as alice, with alice as the admin, so whatever happens for the rest of the test is dubious re: access control